cpp/src/lists/copying/scatter_helper.cu

```diff
@@ -0,0 +1,499 @@
+        auto string_chars_ptr =
+          child_strings_column.child(cudf::strings_column_view::chars_column_index)
+            .template data<char>();
+
+        auto strings_offset = lists_offsets_ptr[row_index] + intra_index;
+        auto char_offset    = string_offsets_ptr[strings_offset];
+        auto char_ptr       = string_chars_ptr + char_offset;
+        auto string_size =
+          string_offsets_ptr[strings_offset + 1] - string_offsets_ptr[strings_offset];
+        return string_view{char_ptr, string_size};
+      });
+
+    // string_views should now have been populated with source and target references.
+
+    auto string_offsets = cudf::strings::detail::child_offsets_from_string_iterator(
+      string_views.begin(), string_views.size(), stream, mr);
+
+    auto string_chars = cudf::strings::detail::child_chars_from_string_vector(
+      string_views, string_offsets->view(), stream, mr);
+    auto child_null_mask =
+      source_lists_column_view.child().nullable() || target_lists_column_view.child().nullable()
+        ? construct_child_nullmask(
+            list_vector, list_offsets, source_lists, target_lists, num_child_rows, stream, mr)
+        : std::make_pair(rmm::device_buffer{}, 0);
+
+    return cudf::make_strings_column(num_child_rows,
+                                     std::move(string_offsets),
+                                     std::move(string_chars),
+                                     child_null_mask.second,            // Null count.
+                                     std::move(child_null_mask.first),  // Null mask.
+                                     stream,
+                                     mr);
+  }
+
+  /**
+   * @brief (Recursively) Constructs a child column that is itself a list column.
+   */
+  template <typename T>
+  std::enable_if_t<std::is_same<T, list_view>::value, std::unique_ptr<column>> operator()(
+    rmm::device_uvector<unbound_list_view> const& list_vector,
+    cudf::column_view const& list_offsets,
+    cudf::lists_column_view const& source_lists_column_view,
+    cudf::lists_column_view const& target_lists_column_view,
+    rmm::cuda_stream_view stream,
+    rmm::mr::device_memory_resource* mr) const
+  {
+    auto source_column_device_view =
+      column_device_view::create(source_lists_column_view.parent(), stream);
+    auto target_column_device_view =
+      column_device_view::create(target_lists_column_view.parent(), stream);
+    auto source_lists = cudf::detail::lists_column_device_view(*source_column_device_view);
+    auto target_lists = cudf::detail::lists_column_device_view(*target_column_device_view);
+
+    auto const num_child_rows{
+      cudf::detail::get_value<size_type>(list_offsets, list_offsets.size() - 1, stream)};
+
+    if (num_child_rows == 0) {
+      // make an empty lists column using the input child type
+      return make_empty_lists_column(
+        source_lists_column_view.child().child(lists_column_view::child_column_index).type(),
```

Comment:
I don't think this handles the case where the LIST column's child type is a nested type, since the `make_empty_lists_column` will simply call `make_empty_column` on the incomplete child type.  Should this be calling `empty_like` or something similar instead?

Reply:
I just hit this bug in the Spark plugin and my tests switching it over to `empty_like` appear to work great. This also is the only place that is calling `make_empty_lists_column` so it could be removed if you switch. I am going to put up a PR anyways just so it does not get lost, and hopefully so I can be unblocked sooner than later, as this is still in draft.

Reply:
@revans2  Sounds good. I'm also looking at other optimizations in this PR so don't let this delay your change.